.github/workflows/dev-build.yml

```diff
@@ -0,0 +1,88 @@
+name: dev-build
+on:
+  push:
+    branches:
+    - master
+    paths-ignore:
+    - 'docker/**'
+    - 'docs/**'
+    - 'notebooks/**'
+    - '.github/markdown.yml'
+  pull_request:
+    types: [ opened, synchronize, reopened, ready_for_review ]
+    branches:
+    - master
+    paths-ignore:
+    - 'docker/**'
+    - 'docs/**'
+    - 'notebooks/**'
+    - '.github/markdown.yml'
+
+jobs:
+  universal:
+    name: Build on node ${{ matrix.node_version }} and ${{ matrix.os }}
+    if: ${{ github.event_name == 'push' || github.event.pull_request.draft == false }}
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 30
+    strategy:
+      matrix:
+        node_version: ['14']
+        os: [ubuntu-latest]
+    steps:
+    - uses: actions/checkout@v2
+  
+    - name: Using Node.js ${{ matrix.node_version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node_version }}
+    - name: Building Pipcook
```

Comment:
How about splitting this into multiple steps:

1. Install Pipcook
2. Build Pipcook
3. Test if the above build work

It displays the cost of every step and we could track it in every build.

Reply:
Done